src/oidcendpoint/oauth2/authorization.py

```diff
@@ -100,27 +290,57 @@ class Authorization(Endpoint):
 
     def __init__(self, endpoint_context, **kwargs):
         Endpoint.__init__(self, endpoint_context, **kwargs)
-        # self.pre_construct.append(self._pre_construct)
         self.post_parse_request.append(self._do_request_uri)
         self.post_parse_request.append(self._post_parse_request)
         self.allowed_request_algorithms = AllowedAlgorithms(ALG_PARAMS)
-        # Has to be done elsewhere. To make sure things happen in order.
-        # self.scopes_supported = available_scopes(endpoint_context)
 
     def filter_request(self, endpoint_context, req):
         return req
 
+    def extra_response_args(self, aresp):
+        return aresp
+
     def verify_response_type(self, request, cinfo):
         # Checking response types
         _registered = [set(rt.split(" ")) for rt in cinfo.get("response_types", [])]
         if not _registered:
             # If no response_type is registered by the client then we'll
-            # code which it the default according to the OIDC spec.
+            # use code.
             _registered = [{"code"}]
 
         # Is the asked for response_type among those that are permitted
         return set(request["response_type"]) in _registered
 
+    def mint_token(self, token_type, grant, session_info, based_on=None):
+        _mngr = self.endpoint_context.session_manager
+        usage_rules = get_usage_rules("authorization_code", self.endpoint_context,
+                                      grant, session_info["client_id"])
+        _exp_in = usage_rules.get("expires_in")
+        if isinstance(_exp_in, str):
+            _exp_in = int(_exp_in)
+
+        token = grant.mint_token(
+            token_type,
+            value=_mngr.token_handler["access_token"](
+                session_info["session_id"],
+                client_id=session_info["client_id"],
+                aud=grant.resources,
+                user_claims=None,
+                scope=grant.scope,
+                sub=session_info["client_session_info"]['sub']
```

Comment:
We shouldn't require the whole `session_info` if we only need the `client_session_info`

Reply:
Also needs session_id. Does it hurt ?